### ISSUE TITLE
Questions on Depth Estimation Example keras: Activation Function, Loss Function

### ISSUE THREAD

**Issue by georgekasa** [keras-team/keras-io#1338]
the issue is for this: https://keras.io/examples/vision/depth_estimation/

1. In the code, the final convolutional layer uses a tanh activation function:
self.conv_layer = layers.Conv2D(1, (1, 1), padding="same", activation="tanh")
However, I noticed that the loss function is defined as
SparseCategoricalCrossentropy with from_logits=True:
The from_logits=True argument suggests that the model's output is
expected to be logits (i.e., without any activation function).

2. Another question I have is about the choice of the tanh activation
function for a depth estimation task. In my understanding, a linear
activation function would seem more appropriate for this type of
problem, as it allows for continuous output values. Could you please
provide some insights into why the tanh activation function was chosen
instead?

**Comment by maitry63** [comment]
Hi @georgekasa, 
About `from_logits=True`, if a model uses: `SparseCategoricalCrossentropy(from_logits=True)`
the final layer should output `raw logits` and should not use an activation function.

 According to, [Keras SparseCategoricalCrossentropy documentation](https://keras.io/api/losses/probabilistic_losses/#sparsecategoricalcrossentropy-class) - "from_logits: Whether to interpret y_pred` as a tensor of logits..."
So, using: `activation="tanh"` together with `from_logits=True` is incorrect because `tanh` limits outputs to `[-1, 1]`, while logits should be unrestricted values.

Depth estimation is a `regression task`, so a `linear activation` is usually the standard choice.
`tanh` is only useful if depth values are normalized to `[-1, 1]`. Since `tanh` always produces values between `-1` and `1`:
If the depth values are not normalized, `tanh` unnecessarily limits the prediction range.

In this Keras example, the model uses a custom regression loss, not `SparseCategoricalCrossentropy`, so `from_logits=True` does not apply. The `tanh` activation is likely used only to keep normalized depth predictions within a fixed range.
